api/users/serializers.py

```diff
@@ -104,6 +104,15 @@ def update(self, instance, validated_data):
             instance.save()
         except ValidationValueError as e:
             raise InvalidModelValueError(detail=e.message)
+        # ValidationError s raised by instance.save() are likely to have e.args[0] be
+        # dict of the form {'field validated': [<Exception>]}
+        except ValidationError as e:
+            if isinstance(e.args[0], dict):
+                value = e.args[0].popitem()[1]
+                if len(value) == 1:
+                    if isinstance(value[0], Exception):
+                        raise InvalidModelValueError(detail=value[0][0])
```

Comment:
OK, I see now; `ValidationErrors` are usually re-raised with a dictionary of errors.

It might be worth adding 

```python
elif isinstance(e.args[0], basestring):
    raise InvalidModelValueError(detail=e.args[0])
```